### PR TITLE
Fix GATK3 overwriting GATK4 executable

### PR DESCRIPTION
**Pull request by apeltzer**
This should hopefully fix https://github.com/bioconda/bioconda-recipes/pull/7138#issuecomment-722100697 

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by apeltzer** [comment]
@BiocondaBot please add label

File: recipes/gatk/build.sh
Comment (mbargull) [inline recipes/gatk/build.sh:25]:
```suggestion
ln -s $PACKAGE_HOME/gatk.py $PREFIX/bin/GenomeAnalysisTK
```
`gatk4` does not have this and previous builds of `gatk` had it as `GenomeAnalysisTK` (i.e., we probably should keep the old name for backward compatibility): https://github.com/bioconda/bioconda-recipes/pull/7138/files#diff-9ade2138ed7598e59317b271854b6193ad573f986cd87e4565ec710795a47292L24

File: recipes/gatk/meta.yaml
Comment (mbargull) [inline recipes/gatk/meta.yaml:31]:
```suggestion
    - GenomeAnalysisTK --version
```

**Comment by apeltzer** [comment]
Thank you @mbargull - just updated this accordingly.

File: recipes/gatk/meta.yaml
Comment (apeltzer) [inline recipes/gatk/meta.yaml:31]:
```suggestion
    - GenomeAnalysisTK --version
    - gatk3 --version
```

**Review by dpryan79 — APPROVED** [review]

**Comment by dpryan79** [comment]
LGTM, approved for merging assuming there are no further proposed changes.

**Comment by mbargull** [comment]
One last thing: https://github.com/bioconda/bioconda-recipes/pull/7138/files also changed `recipes/gatk/3.6`. Can you include the same changes for that one too?

(And then let's merge quickly :).)

File: recipes/gatk/3.6/meta.yaml
Comment (apeltzer) [inline recipes/gatk/3.6/meta.yaml:32]:
```suggestion
    - "gatk3 --version"
```

**Comment by apeltzer** [comment]
All done :-)

**Review by mbargull — APPROVED** [review]
LGTM, thanks @apeltzer!

**Comment by apeltzer** [comment]
I'll wait for tests to pass and will then merge

**Comment by mbargull** [comment]
@BiocondaBot, please fetch artifacts

**Comment by mbargull** [comment]
@BiocondaBot, please merge

**Pull request merged**
